### PR TITLE
[Merged by Bors] - refactor: Avoid `Rat` internals in the definition of `Field`

### PR DESCRIPTION
**Pull request by YaelDillies**
Soon, there will be `NNRat` analogs of the `Rat` fields in the definition of `Field`. `NNRat` is less nicely a structure than `Rat`, hence there is a need to reduce the dependency of `Field` on the internals of `Rat`.

This PR achieves this by restating `Field.ratCast_mk'` in terms of `Rat.num`, `Rat.den`. This requires fixing a few downstream instances.

Reduce the diff of #11203.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/FieldTheory/IsAlgClosed/AlgebraicClosure.lean [hunk 1]
```diff
@@ -424,29 +421,23 @@ def algEquivAlgebraicClosureAux :
...
-      simp only [map_ratCast, map_natCast, map_mul, map_intCast, AlgEquiv.commutes,
-        AlgEquiv.apply_symm_apply]
-      apply Field.ratCast_mk
-    exists_pair_ne := ⟨e.symm 0, e.symm 1, fun w => zero_ne_one ((e.symm).injective w)⟩
-    mul_inv_cancel := fun a w => by
-      apply_fun e
-      simp_rw [map_mul, e.apply_symm_apply, map_one]
-      exact mul_inv_cancel ((AddEquivClass.map_ne_zero_iff e).mpr w)
-    inv_zero := by simp }
+-- Those two instances are copy-pasta from the analogous instances for `SplittingField`
```
Comment (fpvandoorn) [inline Mathlib/FieldTheory/IsAlgClosed/AlgebraicClosure.lean:424]:
```suggestion

-- Those two instances are copy-pasta from the analogous instances for `SplittingField`
```

**Review by fpvandoorn — APPROVED** [review]
Thanks, LGTM.

**Comment by fpvandoorn** [comment]
bors merge
bors d+

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**